test/unit/BondingManager.js

```diff
@@ -1695,6 +1696,139 @@ describe("BondingManager", () => {
         })
     })
 
+    describe("transferBond", () => {
+        let transcoder0
+        let transcoder1
+        let delegator
+        let delegator2
+        const currentRound = 100
+        const ZERO_ADDRESS = ethers.constants.AddressZero
+
+        beforeEach(async () => {
+            transcoder0 = signers[0]
+            transcoder1 = signers[1]
+            delegator = signers[3]
+            delegator2 = signers[4]
+
+            await fixture.roundsManager.setMockBool(functionSig("currentRoundInitialized()"), true)
+            await fixture.roundsManager.setMockBool(functionSig("currentRoundLocked()"), false)
+            await fixture.roundsManager.setMockUint256(functionSig("currentRound()"), currentRound - 1)
+            await bondingManager.connect(transcoder0).bond(1000, transcoder0.address)
+            await bondingManager.connect(transcoder0).transcoder(5, 10)
+            await bondingManager.connect(transcoder1).bond(2000, transcoder1.address)
+            await bondingManager.connect(transcoder1).transcoder(5, 10)
+            await fixture.roundsManager.setMockUint256(functionSig("currentRound()"), currentRound)
+        })
+
+        describe("caller is transferring bond to another address", () => {
+            beforeEach(async () => {
+                await bondingManager.connect(delegator).bond(2000, transcoder0.address)
+                await fixture.roundsManager.setMockUint256(functionSig("currentRound()"), currentRound + 2)
+            })
+
+            describe("receiver is not bonded", () => {
+                it("should transfer the bond to receiver", async () => {
+                    const startDelegatedAmountD1 = (await bondingManager.getDelegator(delegator.address))[0]
+                    const startDelegatedAmountD2 = (await bondingManager.getDelegator(delegator2.address))[0]
+
+                    expect(startDelegatedAmountD1).to.equal(2000)
```

Comment:
Since we're checking `bondedAmount` suggest using that terminology here to avoid confusion with the `delegatedAmount` field which is different.

Reply:
oops 🤦🏻‍♂️